### PR TITLE
refactor(fmgc): fix asobo flight plan syncing issues

### PR DESCRIPTION
**Pull request by 2hwk**
Fixes flight plan syncing from world map issues in Experimental branch where runway symbol would not be displayed on the ND.

## Summary of Changes

Refactor of index values to properly set originRunwayIndex and destinationRunwayIndex, and to use these index values to retrieve onewayRunway objects where needed.

## Additional context

Discord username (if different from GitHub):

## Testing instructions

You will need to test several permutations of the following:

1. Loading in from the world map using MSFS flight planning
2. Manual entry in the MCDU
3. Loading flight plan from SimBrief
4. Loading flight plan from a .PLN save file (saved route from the world map menu)

AND

1. Flight plan with no SID/STAR but departing/arriving to a runway, test at least between 2 different routes*
2. Flight plan with SID but not STAR, test at least between 2 different routes
3. Flight plan with both SID and STAR, test at least between 2 different routes
4. Flight plan with STAR and no approach.

Ensure that the correct SID/STAR is selected (where applicable) the correct approach is displayed on both the MCDU and ND. 

Known Issues: 

Loading with no SID/STAR results in origin and destination runways not being entered - bug with returned values from the sim, no fix.

Sometimes old flight plan will reload from the previous flight if the flight was not started/fully completed. Workaround this by reentering a new Origin/Dest pair before returning to the main menu for now. 

Spawning in mid-air will not show a flight plan. Not addressed by this PR.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_AvailableArrivalsPage.js
Comment (2hwk) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_AvailableArrivalsPage.js]: Fixed

**Review by tracernz — APPROVED** [review]

**Comment by marcman86** [comment]
Quality Assurance Tester Report
Discord:marcman86#4907
Object of testing: #6067
Tier of Testing: 1
Date: 30/10/2021

Testing Process:
- Select a MSFS map flightplan EDDB/LIML (IFR high-altitude airways)
- Dep: ODLU1N 25L, Arr: OSKO1A 36 ILS 36-Y
- Spawned C&D at EDDB
- Start up aircraft
- Checked INIT > shows EDDB/LIML
- Checked flight plan  (Image1, see negatives)
- Checked ILS 36-Y OSKO1A with MCDU arrival input > here waypoint 1313 is not imported, but F-PLN disco. after COD (Image2)

Negatives: 
- CRZ FL is not synced
- F-PLN disco. after SID
- Wrong waypoint in STAR

Testing Results: Not passed
Conclusions: Same problems as of before the PR. F-PLN is fixable with the MCDU. but the PR aims for the right import from the beginning (my understanding). Then it dont works for me.

Hope this helps :-)

Media:


Image1


Image2

**Comment by 2hwk** [comment]
> Quality Assurance Tester Report Discord:marcman86#4907 Object of testing: #6067 Tier of Testing: 1 Date: 30/10/2021
> 
> Testing Process:
> 
> * Select a MSFS map flightplan EDDB/LIML (IFR high-altitude airways)
> * Dep: ODLU1N 25L, Arr: OSKO1A 36 ILS 36-Y
> * Spawned C&D at EDDB
> * Start up aircraft
> * Checked INIT > shows EDDB/LIML
> * Checked flight plan  (Image1, see negatives)
> * Checked ILS 36-Y OSKO1A with MCDU arrival input > here waypoint 1313 is not imported, but F-PLN disco. after COD (Image2)
> 
> Negatives:
> 
> * CRZ FL is not synced
> * F-PLN disco. after SID
> * Wrong waypoint in STAR
> 
> Testing Results: Not passed Conclusions: Same problems as of before the PR. F-PLN is fixable with the MCDU. but the PR aims for the right import from the beginning (my understanding). Then it dont works for me.
> 
> Hope this helps :-)
> 
> Media:
> 
>  Image1
> 
>  Image2

The fix in this PR is targeted at solving approaches/destinations not being shown on the ND on load in, sorry for the confusion, this is not a solution for all flight plan issues in general.

**Comment by John-UK** [comment]
Quality Assurance Tester Report
Discord:john P Maguire#9027
Object of testing: #6067
Tier of Testing: 2
Date: 07/11/21

Testing Process:

From previous QA report rechecked

Select a MSFS map flightplan EDDB/LIML (IFR high-altitude airways)
Dep: ODLU1N 25L, Arr: OSKO1A 36 ILS 36-Y
Spawned C&D at EDDB
See discord for full details. Not sure CRZ FL would be synced. Where would world map record that data if it does? I did not see or expect cruise level to be prefilled. Neither alternative destination, flight number or CI.
The discontinuity seen was recreated by not using a via. When a via COD was selected a full sequence to the RW was seen. This disco was seen in both the experimental and this PR.

Flights using the world map (EGJJ-EGLL EGLL-LFPG, EHAM-EGLL) were checked. With SID without, with STAR without. Each was imported into this PR and checked against the same as seen when imported into exp. 

A flight was created in LittleNavMap (EHAM-EGLL) from EGAM stand 22 to EGLL RW27 with full SID and STAR.  See discord for details. The world map imported the LNM flight plan perfectly. 

The aircraft spawned at the correct gate C&D. Once on external power the flight plan was then available in the MCDU. Both the PR and experimental imported in the same plan. Both had the same data in the MCDU.  The route was flown using AP from 600 feet to full auto land in PR and exp. No differences noted. 

Test result: Pass

Negatives: None noted for this PR. The LAM weirdness is also in the exp version. 

Comments: For those who do not use SIMBRIEF. LNM is an ideal alternative. It is also a great interface to see the SIDs, STARs etc. Could be worth including in documentation, if not already there.

**Comment by 2hwk** [comment]
Will be retargeted at master and merged after #4953

**Comment by lolisec** [comment]
Quality Assurance Trainee Report

Discord : Unfriended#6017
Object of testing: #6067 
Tier of Testing  : 1
Date  : (11/19/2021)

Testing Process: 
1. Spawn at KPSC C&D
2. Turned on both batteries + external power
3. Set the ND to PLAN mode.
3. Checked the MCDU fplan-page and scrolled through each waypoint.

Negatives:       
N/A

Testing Results: 
Passed

Conclusions: 
Everything seems fixed. No issues were found. It worked perfectly.

image

**Comment by 2hwk** [comment]
QA Passed ☑
Retargeted at master, **do not merge into master** until 4953 is merged.

**Pull request merged**
